packages/forms/src/widgets/ListViewWidget/ListViewWidget.js

```diff
@@ -78,11 +90,16 @@ class ListViewWidget extends React.Component {
 	}
 
 	componentWillReceiveProps(nextProps) {
-		const options = this.props.options || {};
-		const nextOptions = nextProps.options || {};
-		if (options.enumOptions !== nextOptions.enumOptions) {
-			const items = getItems(nextOptions, nextProps.value, this);
-			this.setState({ items, displayedItems: items });
+		const options = this.props.options;
+		const nextOptions = nextProps.options;
+		if (!areOptionsEqual(options, nextOptions)) {
+			const items = getItems(nextOptions, this.props.value, this);
+			this.setState({
+				items,
+				displayedItems: items,
+				searchCriteria: null,
+				toggleAllChecked: items.length === items.filter(i => i.checked).length,
```

Comment:
* `items.every(i => i.checked)` so it stops on first element that doesn't match
* can you mutualise the `{ item, displayedItems, toggleAllChecked }` from here + constructor ? Perhaps a `this.getDerivedStateFromOptions(newOptions )`

```
	/**
	* This computes new derived state when options change
	* Compute new items
	* Reset search and displayedItems
	* Update toggleAll status
	*/
	getDerivedStateFromNewOptions(options) {
		const items = getItems(options, this.props.value, onItemChange.bind(this));
		return {
			items,
			searchCriteria: null,
			displayedItems: items,
			toggleAllChecked: items.every(i => i.checked),
		};
	}
```

What do you think ?

Reply:
👍